app/controllers/application_controller.rb

```diff
@@ -1,10 +1,166 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do
+    erb :index
+  end
+
+
+  get '/users/:id' do 
+    @user = User.find_by(id: params[:id])
+    erb :'users/show'
+  end
+
+  get '/signup' do
+   if !session[:user_id]
+      erb :'users/signup'
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/signup' do
+
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect to '/signup'
+    else
+      @user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
+      @user.save
+      session["user_id"] = @user.id
+      redirect to '/tweets'
+    end
+  end
+
+
+  get '/login' do 
+    if !session[:user_id]
+      erb :'users/login'
+    else
+      redirect '/tweets'
+    end
+  end
+
+
+ post '/login' do
+    user = User.find_by(:username => params[:username])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect to '/signup'
+    end
+  end
+
+  get '/logout' do
+    if session[:user_id] != nil
+      session.destroy
+      redirect to '/login'
+    else
+      redirect to '/'
+    end
+  end
+
+#TWEETS
+  
+  get '/tweets/new' do
+    if !session[:user_id]
+      redirect to :'login'
+    else
+      erb :'tweets/create_tweet'
+    end
+  end
+
+  get '/tweets' do
+    if session[:user_id]
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect to :'/login'
+    end
+  end
+
+  post '/tweets' do
+
+    if params[:content] == ""
+      redirect to '/tweets/new'
+    else
+      @tweet = Tweet.new(:content => params[:content])
+      @tweet.user_id = session[:user_id]
+      @tweet.save
+      redirect to "/tweets/#{@tweet.id}"
+    end
+  end
+
+  get '/tweets/:id' do 
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` before allowing them to edit?